### PR TITLE
[DOCS] Document index.load_fixed_bitset_filters_eagerly

### LINKED ISSUE #14903: Document index.load_fixed_bitset_filters_eagerly
FixedBitSetFilterCache is a data structure that is loaded eagerly in memory (by default) to support nested query/filter and nested aggregations. However, the problem is that it can cause it to use too much heap for it is loaded for all nested fields (regardless of whether these fields are being used). To prevent this from happening, a common configuration workaround is to set `index.load_fixed_bitset_filters_eagerly: false` in the yml of the nodes and restart them to prevent the nodes from running OOM when attempting to eagerly load the fixedbitsets. 

At this point, I have seen enough occurrences of this in the field which necessitates the setting of `index.load_fixed_bitset_filters_eagerly: false` to prevent the nodes from running out of memory.  It may make sense to start documenting this setting unless there is a better solution we are already planning :)

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the `index.load_fixed_bitset_filters_eagerly` setting.

#### Context
Prior to 2.3, users would frequently set `index.load_fixed_bitset_filters_eagerly` to `false` to avoid memory errors for mappings containing a large number of nested fields.

The need for that workaround was largely reduced in 2.3+ with #15989. That PR introduced `index.mapping.nested_fields.limit` which limits the number of nested fields. For that reason, I avoided referencing the workaround or other heuristics in this definition.

#### Related Tickets
Resolves #14903

### CODE REVIEW

File: docs/reference/index-modules.asciidoc [hunk 1]
```diff
@@ -83,6 +83,11 @@ WARNING: Expert only. Checking shards may take a lot of time on large indices.
     than the `index.number_of_shards` unless the `index.number_of_shards` value is also 1.
     See <<routing-index-partition>> for more details about how this setting is used.
 
+[[load-fixed-bitset-filters-eagerly]] `index.load_fixed_bitset_filters_eagerly`::
+
+    Indicates where <<query-filter-context, cached filters>> are pre-loaded for
```
Comment (jpountz) [inline docs/reference/index-modules.asciidoc]: did you mean s/where/whether/?

Comment (jrodewig) [reply docs/reference/index-modules.asciidoc]: 🤦‍♂️  I did mean 'whether.' Thanks for catching that @jpountz. Fixed with c8f5200.

**Review by jimczi — APPROVED** [review]

**Review by jpountz — APPROVED** [review]

**Comment by mayya-sharipova** [comment]
@jrodewig Thanks for updating documentation. Would you mind to backport this change to earlier versions as well (at least 5.6 and 6.7). We had an SDH issue where people were wondering if this setting is available in earlier versions as well.

**Comment by jrodewig** [comment]
@mayya-sharipova  Sure. I've backported to 5.6 and all versions since. Let me know if you need anything else.

**Pull request merged**